### DISCUSSION TITLE
How to type an object that contains any or all of types generated by Prisma

### DISCUSSION THREAD

**Discussion by dylanjt** [prisma/prisma#17920] [Q&A]
Hi everyone,

I am trying to describe an object that can contain many or all of the types generated by Prisma:

```ts
import type * as p from "@prisma/client";

type BuilderModels = Partial<{
  [key in p.Prisma.ModelName]: p[key][];
}>;
```

This does not work, because p is a namespace. I tried adding the `typeof` operator, but it seems to _remove_ the actual types, and only contains the generated consts representing enums in the schema.

```ts
import type * as p from "@prisma/client";

type BuilderModels = Partial<{
  [key in p.Prisma.ModelName]: typeof p[key][];
}>;
```

Is it possible to describe such a type? Is it possible to access my generated types via index?

**Comment by nurul3101** [comment]
Hello @dylanjt 👋 
Can you let me know what is your main use case for creating this object?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
